### DISCUSSION TITLE
initQpStatefulProperties() for initial value setting

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#26484] [Q&A General] [answered]
All,
I am trying to a do loop summation calculation like shown below in C++.
```
float sum = 0.5;
    for (int i = 0; i <= 10; i++)
    {
        sum = sum + 20;
    }
```
I programed a header file “MaterialPropertyOld.h” and a source file “MaterialPropertyOld.C” (please see attached).

The problem is that the initial vale of 0.5 is not read by 
MaterialPropertyOld::initQpStatefulProperties()
{  _sum[_qp] = _initial_sum;}

Please help me with how to set an initial value of 0.5.
S. Thomas
[MaterialPropertyOld_h.pdf](https://github.com/idaholab/moose/files/13837080/MaterialPropertyOld_h.pdf)
[MaterialPropertyOld_C.pdf](https://github.com/idaholab/moose/files/13837083/MaterialPropertyOld_C.pdf)

**Comment by GiudGiud** [comment]
Hello

This should have been the way to do this. 
How did you check that it's not working?
In your input file did you override the default for intiial_sum?

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
"VectorPostprocessors" prints: sum = 20, 40, 60, ... instead of 20.5, 40.4, 60.5, ...
I have set "initial_sum = 0.5" in the input file, too.

Do you have an example(s) of using initQpStatefulProperties()?
S. Thomas

**Reply by GiudGiud** [reply]
In tensor mechanics I looked at PorosityFromStrain that uses a similar pattern.
Maybe you ll see what s different between the two

**Reply by styyokuda** [reply]
Thank you, Guillaume.
Could you be please more specific where I can find this example?
I cannot find it in /projects/moose/modules/tensor_mechanics/
S. Thomas

**Reply by GiudGiud** [reply]
see /projects/moose/modules/tensor_mechanics/src/materials/PorosityFromStrain.C

**Reply by styyokuda** [reply, marked as answer]
Thank you, Guillaume! It is working now. :)
